mmf/trainers/core/training_loop.py

```diff
@@ -164,10 +164,25 @@ def run_training_epoch(self) -> None:
 
     def run_training_batch(self, batch: Dict[str, Tensor], loss_divisor: int) -> None:
         report = self._forward(batch)
+        if self.training_config.exit_on_nan_losses:
+            self._check_nan_losses(report)
         loss = extract_loss(report, loss_divisor)
```

Comment:
You can check afterwards on the loss as well, no? That would anyways be nan. I think ideal place for this check would be just before actual backward happens.

Reply:
I was deliberately putting it over `report` rather than loss and before `extract_loss`, so that it would check for each loss before reduction, and provide some clue on which loss and which process lead to NaNs. But I can change it to just check the output of `extract_loss` if you feel that's better 😄